### ISSUE TITLE
Fedora won't converge

### ISSUE THREAD

**Issue by mjulian** [neillturner/kitchen-puppet#145]
`dnf` on Fedora 23 does not have the `localinstall` option causing the setup phase to bail:

```       Installing Puppet Collections on fedora
-----> sudo -E yum -y localinstall http://yum.puppetlabs.com/puppetlabs-release-pc1-fedora-23.noarch.rpm
       Yum command has been deprecated, redirecting to '/usr/bin/dnf -y localinstall http://yum.puppetlabs.com/puppetlabs-release-pc1-fedora-23.noarch.rpm'.
       See 'man dnf' and 'man yum2dnf' for more information.
       To transfer transaction metadata from yum to DNF, run:
       'dnf install python-dnf-plugins-extras-migrate && dnf-2 migrate'

       No such command: localinstall. Please use /usr/bin/dnf --help
       It could be a DNF plugin command, try: "dnf install 'dnf-command(localinstall)'"
       Yum command has been deprecated, redirecting to '/usr/bin/dnf -y install puppet-agent'.
       See 'man dnf' and 'man yum2dnf' for more information.
       To transfer transaction metadata from yum to DNF, run:
       'dnf install python-dnf-plugins-extras-migrate && dnf-2 migrate'

Fedora 23 - x86_64 - Updates                    4.3 MB/s |  25 MB     00:05    :-- ETA
       Last metadata expiration check performed 0:00:15 ago on Wed Nov 16 22:53:23 2016.
       No package puppet-agent available.
       Error: Unable to find a match.
```

Installing the repo and `puppet-agent` package manually on Fedora works just fine.

One potential fix is to move Fedora from the RH-based conditional to its own conditional and use `dnf install` instead of relying on `yum` to redirect to `dnf` (which doesn't appear to work well). Another option could be to drop the `localinstall` argument to `yum`, as it's not necessary (`yum install` does the right thing when given a package on disk).

**Comment by neillturner** [comment]
dropping localinstall argument is simplest. Will this work on all redhat systems or is it just for fedora and i then need to identify it is fedora somehow

**Comment by mjulian** [comment]
I know for certain it works on RHEL/CentOS 6 and 7, but not positive about 5.x.

**Comment by neillturner** [comment]
i wouldn't worry about 5.x now so i'll take  localinstall out

**Comment by neillturner** [comment]
fixed in version 1.45.2
